includes/class-sensei-analysis-overview-list-table.php

```diff
@@ -142,6 +147,98 @@ function get_columns() {
 
 		return $this->columns;
 	}
+
+	/**
+	 * Get total average progress for all the courses.
+	 *
+	 * @since  4.3.0
+	 * @access private
+	 *
+	 * @return string average_progress total average progress value and N/A if not applicable.
+	 */
+	private function get_total_average_progress_for_courses_report() {
+		// Get all courses ids.
+		$courses_ids = $this->get_all_courses_ids();
+
+		// Get all students for courses.
+		$students = Sensei()->course->get_students_grouped_by_courses( $courses_ids );
+
+		// Get students grouped by courses.
+		$students_by_courses = array();
+		foreach ( $students as $student ) {
+			$students_by_courses[ $student->course_id ][] = $student->student_id;
+		}
+
+		// Get Lessons by courses.
+		$lessons = Sensei()->course->get_lessons_grouped_by_courses( $courses_ids );
+
+		// Get Lessons grouped by courses.
+		$lessons_by_courses = array();
+		foreach ( $lessons as $lesson ) {
+			$lessons_by_courses[ $lesson->course_id ][] = $lesson->lesson_id;
+		}
+
+		// Calculate total average progress for all the courses.
+		$average_course_progress_total = 0;
+		foreach ( $courses_ids as $course_id ) {
+			// Calculate average progress per course.
+
+			// First step: Get total max value by multiplying students and lessons.
+			if ( ! is_array( $students_by_courses[ $course_id ] ) || ! is_array( $lessons_by_courses[ $course_id ] ) ) {
+				continue;
+			}
+			$lessons_students_max_value = ( count( $students_by_courses[ $course_id ] ) * count( $lessons_by_courses[ $course_id ] ) );
+
+			// Second step: Get completed lessons in the course.
+			$completed_lessons_in_course = Sensei()->lesson->get_completed_lessons_for_students_and_lessons( $lessons_by_courses[ $course_id ], $students_by_courses[ $course_id ] );
+			if ( $completed_lessons_in_course && is_array( $completed_lessons_in_course ) ) {
+				$average_course_progress_total += ( count( $completed_lessons_in_course ) / $lessons_students_max_value * 100 );
+			}
+		}
+		// If total value for all the courses is zero, return N/A.
+		if ( ! $average_course_progress_total ) {
+			return __( 'N/A', 'sensei-lms' );
```

Comment:
So we want to distinguish between 0% and N/A, this would require returning potential -1 or something like that so that we know that it's N/A which would add more if-ology to the place where we use it. But if you are strong about this, I don't mind changing it like that.